x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/PlanExecutor.java

```diff
@@ -66,6 +68,12 @@ public void searchSource(Configuration cfg, String sql, List<SqlTypedParamValue>
             if (exec instanceof EsQueryExec) {
                 EsQueryExec e = (EsQueryExec) exec;
                 listener.onResponse(SourceGenerator.sourceBuilder(e.queryContainer(), cfg.filter(), cfg.pageSize()));
+            } else if (exec instanceof LocalExec) {
+                listener.onFailure(new PlanningException("Cannot generate a query DSL for an SQL query that either " +
+                    "its WHERE clause evaluates to FALSE or doesn't operate on a table (missing a FROM clause), sql statement: {}"));
+            } else if (exec instanceof CommandExec) {
+                listener.onFailure(new PlanningException("Cannot generate a query DSL for a special SQL command " +
+                    "(e.g.: DESCRIBE, SHOW), sql statement: {}"));
             } else {
                 listener.onFailure(new PlanningException("Cannot generate a query DSL for {}", sql));
```

Comment:
Would be a good idea to improve the visibility of the query being logged? "Cannot generate a query DSL for {}" would look like `Cannot generate a query DSL for select * from test` for a query. My suggestion is to either use square brackets for the query itself - `Cannot generate a query DSL for [select * from test]` - or be in line with the newly added error messages or those newly added error messages to be consistent with the existent one. Either way: the style of the error message should be consistent and the query itself to be more visible.